loader/get_environment.c

```diff
@@ -111,28 +112,30 @@ bool is_high_integrity() {
 }
 
 char *loader_getenv(const char *name, const struct loader_instance *inst) {
-    char *retVal;
-    DWORD valSize;
-
-    valSize = GetEnvironmentVariableA(name, NULL, 0);
+    int name_utf16_size = MultiByteToWideChar(CP_UTF8, 0, name, -1, NULL, 0);
+    if (name_utf16_size <= 0) {
+        return NULL;
+    }
+    wchar_t *name_utf16 = (wchar_t *)loader_stack_alloc(name_utf16_size * sizeof(wchar_t));
+    MultiByteToWideChar(CP_UTF8, 0, name, -1, name_utf16, name_utf16_size);
 
-    // valSize DOES include the null terminator, so for any set variable
+    DWORD val_size = GetEnvironmentVariableW(name_utf16, NULL, 0);
+    // val_size DOES include the null terminator, so for any set variable
     // will always be at least 1. If it's 0, the variable wasn't set.
-    if (valSize == 0) return NULL;
-
-    // Allocate the space necessary for the registry entry
-    if (NULL != inst && NULL != inst->alloc_callbacks.pfnAllocation) {
-        retVal = (char *)inst->alloc_callbacks.pfnAllocation(inst->alloc_callbacks.pUserData, valSize, sizeof(char *),
-                                                             VK_SYSTEM_ALLOCATION_SCOPE_COMMAND);
-    } else {
-        retVal = (char *)malloc(valSize);
+    if (val_size == 0) {
+        return NULL;
     }
 
-    if (NULL != retVal) {
-        GetEnvironmentVariableA(name, retVal, valSize);
-    }
+    wchar_t *val = (wchar_t *)loader_stack_alloc(val_size * sizeof(wchar_t));
+    GetEnvironmentVariableW(name_utf16, val, val_size);
```

Comment:
Original code with `GetEnvironmentVariableA` does not check return value twice. I am not sure that it really necessary to check return value of winapi function like `GetEnvironmentVariable`/`MultiByteToWideChar` twice because we pass size that calculated by them in the previous step. But I will add additional check that size is still the same.